### PR TITLE
Add remote cache support

### PR DESCRIPTION
**Pull request by nizamutdinovb**
Modified pipeline to use remote cache located in Google Cloud Storage with Google credentials.
Google credentials are stored in GitHub organization secrets.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -63,6 +73,8 @@ jobs:
             --spawn_strategy=local \
             -c dbg \
             --strip="never" \
+            --remote_cache=$GOOGLE_REMOTE_CACHE \
+            --google_credentials=$GOOGLE_APPLICATION_CREDENTIALS \
             :eventuals
```
Comment (rjhuijsman) [inline .github/workflows/build_and_test.yml:78]: Can we also add `--verbose_failures`? If the credentials stop working at some point (or just aren't there) `bazel` will give a very limited error, with `--verbose_failures` it's much clearer.

**Review by rjhuijsman — APPROVED** [review]
Nice!

Approved the PR with a small comment, go ahead and merge it when you've addressed that!

**Comment by FolMing** [comment]
Please squash everything into 1 commit before merging into main. Thank you!

**Comment by nizamutdinovb** [comment]
> Please squash everything into 1 commit before merging into main. Thank you!

Yes, GitHub PR does it automatically when merging the PR with option "Squash and Merge".

**Pull request merged**